test/tomo/operators/spect_trafo_test.py

```diff
@@ -0,0 +1,123 @@
+    """Test discrete SPECT transform."""
+
+    # Geometry
+    det_nx_pix = 32
+    det_ny_pix = 32
+    det_nx_mm = 4
+    det_radius = 200
+    n_proj = 16
+    det_param = det_nx_mm * det_nx_pix
+    dpart = odl.uniform_partition([-det_param, -det_param],
+                                  [det_param, det_param],
+                                  [det_nx_pix, det_ny_pix])
+
+    apart = odl.uniform_partition(0, 2 * np.pi, n_proj)
+    geometry = odl.tomo.geometry.spect.ParallelHoleCollimatorGeometry(
+        apart, dpart, det_rad=det_radius)
+
+    # Create a discrete domain and a phantom as an element of the domain
+    domain = odl.uniform_discr([-20] * 3, [20] * 3, [det_nx_pix] * 3)
+
+    vol = odl.util.phantom.shepp_logan(domain, True)
+    # Create a SPECT projector
+    projector = AttenuatedRayTransform(
+        domain, geometry, attenuation=None, impl='niftyrec')
+
+    # Calculate projection and back-projection
+    proj = projector(vol)
+    backproj = projector.adjoint(proj)
+
+    assert all_equal(proj.shape, (n_proj, det_nx_pix, det_ny_pix))
+    assert all_equal(backproj.shape, vol.shape)
+    assert proj in projector.range
+    assert backproj in projector.domain
+
+    # Adjoint test:  Verified the identity <Ax, Ax> = <A^* A x, x>
+    result_AxAx = proj.inner(proj)
+    result_xAtAx = backproj.inner(vol)
+    # Accept 1% errors
+    places = 2
+    assert almost_equal(result_AxAx, result_xAtAx, places=places)
+
+
+@skip_if_no_niftyrec
+def test_attenuation():
+    det_nx_pix = 16
+    det_ny_pix = 16
+    det_nx_mm = 0.4
+    det_radius = 20
+    n_proj = 1
+    det_param = det_nx_mm * det_nx_pix
+    dpart = odl.uniform_partition([-det_param, -det_param],
+                                  [det_param, det_param],
+                                  [det_nx_pix, det_ny_pix])
+
+    apart = odl.uniform_partition(0, 2 * np.pi, n_proj)
+    geometry = odl.tomo.geometry.spect.ParallelHoleCollimatorGeometry(
+        apart, dpart, det_rad=det_radius)
+
+    # Create a discrete domain and a phantom as an element of the domain
+    domain = odl.uniform_discr([-1] * 3, [1] * 3, [det_nx_pix] * 3)
+
+    vol = np.zeros((16, 16, 16))
+    vol[0, 0, 0] = 100
+
+    vol_element = domain.element(vol)
+    projector = AttenuatedRayTransform(
+        domain, geometry, attenuation=None, impl='niftyrec')
+    proj_no_att = projector(vol_element)
+    # Accept 0.1 % error
+    places = 3
+    assert odl.util.testutils.almost_equal(np.max(proj_no_att), 100,
+                                           places=places)
+
+    attenuation = np.zeros((16, 16, 16))
+    attenuation[0, 0, 0] = 0.2
+    projector_att = AttenuatedRayTransform(
+        domain, geometry, attenuation=attenuation, impl='niftyrec')
+    proj_att = projector_att(vol_element)
+
+    value = np.sum(vol * np.exp(-np.sum(attenuation)))
+    assert odl.util.testutils.almost_equal(np.max(proj_att), value,
+                                           places=places)
+
```

Comment:
Another note for these tests. Should'nt we expect the following to hold:

```
det_nx_pix = 16
det_ny_pix = 16
det_nx_mm = 0.4
det_radius = 20
n_proj = 1
det_param = det_nx_mm * det_nx_pix

domain = odl.uniform_discr([-1] * 3, [1] * 3, [det_nx_pix] * 3)

dpart = odl.uniform_partition([-det_param, -det_param],
                              [det_param, det_param],
                              [det_nx_pix, det_ny_pix])

apart = odl.uniform_partition(0, 2 * np.pi, n_proj)

geometry1 = odl.tomo.geometry.spect.ParallelHoleCollimatorGeometry(
    apart, dpart, det_rad=det_radius)
geometry2 = odl.tomo.geometry.Parallel3dAxisGeometry(
    apart, dpart)
    
projector1 = AttenuatedRayTransform(
    domain, geometry, impl='niftyrec')
projector2 = RayTransform(
    domain, geometry, impl='astra_cuda')
    
data = odl.util.shepp_logan(domain)

assert projector1(data) == projector2(data)
```

Perhaps we should add a slow test that verifies that this is the case? You could add `AttenuatedRayTransform` with the right parameters [here](https://github.com/odlgroup/odl/blob/3543720d91841c602bfebf94bdbe26986a6f3718/test/largescale/tomo/ray_transform_slow_test.py#L37-L56).